kinisi/yeh_hummer.py

```diff
@@ -0,0 +1,328 @@
+        # Convert box lengths to inverse values
+        L_values = self.box_lengths.values
+        inv_L = 1.0 / L_values
+
+        # Get diffusion values and errors
+        D_values = self.diffusion.values
+        D_errors = np.sqrt(self.diffusion.variances)
+
+        return inv_L, D_values, D_errors
+
+    def _slope_to_viscosity(self, slope):
+        """Convert slope to viscosity using Yeh-Hummer relation."""
+        # slope = (k_B * T * xi) / (6 * pi * eta)
+        # eta = (k_B * T * xi) / (6 * pi * slope)
+
+        k_B_T = sc.to_unit(self.k_B * self.temperature, 'J')
+
+        # slope has units of [diffusion] / [1/length] = [diffusion] * [length]
+        # diffusion is cm^2/s, box_lengths is Å, so slope * diffusion.unit / (1/box_lengths.unit)
+        # This gives us (cm^2/s) / (1/Å) = cm^2/s * Å = cm^2 * Å / s
+        slope_with_units = slope * self.diffusion.unit / (1 / self.box_lengths.unit)
+        slope_SI = sc.to_unit(slope_with_units, 'm^3/s')
+
+        eta = (k_B_T * self.xi_cubic) / (6 * np.pi * slope_SI)
+        return sc.to_unit(eta, 'Pa*s')
+
+    def _viscosity_to_slope(self, eta):
+        """Convert viscosity to slope for fitting."""
+        slope = (self.k_B * self.temperature * self.xi_cubic) / (6 * np.pi * eta)
+
+        # Convert back to data units
+        target_unit = self.diffusion.unit * self.box_lengths.unit
+        return sc.to_unit(slope, target_unit).value
+
+    def max_likelihood(self):
+        """Find maximum likelihood parameters with better initial guess for YehHummer."""
+        # Use linear fit for initial parameters
+        inv_L, D_values, D_errors = self._prepare_data_for_fit()
+
+        def linear_func(x, a, b):
+            return a - b * x
+
+        popt, _ = curve_fit(
+            linear_func,
+            inv_L,
+            D_values,
+            sigma=D_errors if np.any(D_errors > 0) else None,
+            p0=[np.max(D_values), (D_values[0] - D_values[-1]) / (inv_L[0] - inv_L[-1])],
+        )
+
+        D_0_init = popt[0]
+        slope_init = popt[1]
+
+        # Convert slope to viscosity
+        eta_init = self._slope_to_viscosity(slope_init).value
+
+        # Use these as initial parameters for optimization
+        x0 = [D_0_init, eta_init]
+
+        from scipy.optimize import minimize
+
+        # Convert bounds to format expected by scipy
+        bounds_scipy = [(b[0].value, b[1].value) for b in self.bounds]
+        result = minimize(self.nll, x0, bounds=bounds_scipy, method='L-BFGS-B')
+
+        # Store results
+        self.data_group['D_0'] = result.x[0] * self.parameter_units[0]
+        self.data_group['viscosity'] = result.x[1] * self.parameter_units[1]
+
+    @property
+    def D_infinite(self):
+        """Return infinite-system diffusion coefficient."""
+        return self.data_group['D_0']
+
+    @property
+    def shear_viscosity(self):
+        """Return estimated shear viscosity."""
+        return self.data_group['viscosity']
+
+    def plot(
```

Comment:
I don't love plotting functions in the library (can lead to technical debt headaches). Can you move this to the tutorial notebook?